### PR TITLE
Fix compiler crash and remove spurious hidden sequence points for switch expression

### LINKED ISSUE #32370: Stepping through switch expression is wrong
The whole statement with the switch expression never gets highlighted as I step forward.

switch-stepping

When fixing this, please also verify that LUT (live unittesting) instrumentation works.

### LINKED ISSUE #32749: Compiler crash on incomplete switch expression
Parsing the following expression causes the parser (or compiler) to crash with an out-of-memory exception

``` c#
(e switch {)
```

### PR DESCRIPTION
**Pull request by gafter**
Fixes #32370

Fix parsing error recovery for incomplete switch expression
Fixes #32749 

See also https://devdiv.visualstudio.com/DevDiv/_workitems/edit/792979 for shiproom purposes.

@dotnet/roslyn-compiler May I please have a couple of reviews?

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs [hunk 1]
```diff
@@ -2130,5 +2130,67 @@ static class C {
...
+  IL_0027:  stloc.3
+  IL_0028:  ldloc.3
+  IL_0029:  callvirt   ""Program Program.Chain()""
+  IL_002e:  stloc.1
+ -IL_002f:  ldloc.1
+  IL_0030:  callvirt   ""Program Program.Chain2()""
+  IL_0035:  pop
+ -IL_0036:  ret
+}
+", sequencePoints: "Program.Main");
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs]: nit: you can generate a friendlier output by using the parameter for source code

**Comment by gafter** [comment]
@jaredpar This fixes a compiler crash and bad debug experience for preview3. What approvals are needed?

**Comment by jcouv** [comment]
It looks like there is a genuine test failure on Linux (index out of range). Marked PR as "personal"

**Comment by gafter** [comment]
I don't know why the Linux test is failing (I suspect a Linux runtime bug), but I'm going to separate the two fixes herein and submit only the crash fix for `dev16.0`

**Comment by jaredpar** [comment]
@gafter 

Looks like a straight bug in our test framework

```
at Roslyn.Test.Utilities.ILValidation.SnippetFromSpan(String[] lines, XmlElement span) in /_/src/Test/Utilities/Portable/Metadata/ILValidation.cs:line 402
   at Roslyn.Test.Utilities.ILValidation.GetSequencePointMarkers(String pdbXml, String source) in /_/src/Test/Utilities/Portable/Metadata/ILValidation.cs:line 360
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.VisualizeIL(MethodData methodData, Boolean realIL, String sequencePoints, String source) in /_/src/Test/Utilities/Portable/CompilationVerifier.cs:line 323
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.VisualizeIL(String qualifiedMethodName, Boolean realIL, String sequencePoints, String source) in /_/src/Test/Utilities/Portable/CompilationVerifier.cs:line 289
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.VerifyILImpl(String qualifiedMethodName, String expectedIL, Boolean realIL, String sequencePoints, String callerPath, Int32 callerLine, Boolean escapeQuotes, String source) in /_/src/Test/Utilities/Portable/CompilationVerifier.cs:line 279
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.VerifyIL(String qualifiedMethodName, String expectedIL, Boolean realIL, String sequencePoints, String callerPath, Int32 callerLine, String source) in /_/src/Test/Utilities/Portable/CompilationVerifier.cs:line 255
   at Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen.PatternTests.SwitchExpressionSequencePoints() in /_/src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs:line 2156
```

**Pull request merged**